core/src/main/scala/org/apache/spark/SparkThrowableHelper.scala

```diff
@@ -135,4 +143,41 @@ private[spark] object SparkThrowableHelper {
   def isInternalError(errorClass: String): Boolean = {
     errorClass == "INTERNAL_ERROR"
   }
+
+  def getMessage(e: SparkThrowable with Throwable, format: ErrorMessageFormat.Value): String = {
+    import ErrorMessageFormat._
+    format match {
+      case PRETTY => e.getMessage
+      case MINIMAL | STANDARD if e.getErrorClass == null =>
+        val jValue = ("errorClass" -> "legacy") ~
+          ("messageParameters" -> JObject(List("message" -> JString(e.getMessage)))) ~
+          ("queryContext" -> JArray(List.empty))
+        compact(render(jValue))
+      case MINIMAL | STANDARD =>
+        val errorClass = e.getErrorClass
+        val message = if (format == STANDARD) {
+          val errorInfo = errorClassToInfoMap.getOrElse(errorClass,
+            throw new IllegalArgumentException(s"Cannot find error class '$errorClass'"))
+          Some(errorInfo.messageFormat)
+        } else None
+        assert(e.getParameterNames.size == e.getMessageParameters.size,
+          "Number of message parameter names and values must be the same")
+        val jValue = ("errorClass" -> errorClass) ~
+          ("errorSubClass" -> Option(e.getErrorSubClass)) ~
+          ("message" -> message) ~
+          ("sqlState" -> Option(e.getSqlState)) ~
+          ("messageParameters" ->
+            JObject((e.getParameterNames zip e.getMessageParameters.map(JString)).toList)) ~
+          ("queryContext" -> JArray(
+            e.getQueryContext.map(c => JObject(
+              "objectType" -> JString(c.objectType()),
+              "objectName" -> JString(c.objectName()),
+              "startIndex" -> JInt(c.startIndex()),
+              "stopIndex" -> JInt(c.stopIndex()),
+              "fragment" -> JString(c.fragment()))).toList)
+          )
+        val rendered = render(jValue)
+        if (format == MINIMAL) compact(rendered) else pretty(rendered)
```

Comment:
shall we always use the pretty JSON output instead od compacted?

Reply:
I supposed that minimal means minimal size of result JSON that will be parsed machinery. What's the use case for the MINIMAL form from your point of view. Do you mean that it could improve readability of golden files?

Reply:
MINIMAL only has payload. It does not contain the message text at all.
So it's needed for the Golden Files (recall that teh original idea was to do machine readable before golden files),
It may be advised for logs - so we do keep stability.
and it may be useful for clients that want to get their messages from another (future) API. For example if they have a localized set of messages.

Reply:
PS: We need an API (table function?) to return error messages.....

Reply:
@srielau Should we output MINIMAL in the compact or pretty form?